### PR TITLE
build fedora 26 packages

### PR DESCRIPTION
**Pull request by andrewhsu**
Just released: https://docs.fedoraproject.org/en-US/Fedora/26/html/Release_Notes/

Signed-off-by: Andrew Hsu <andrewhsu@docker.com>

### CODE REVIEW

File: rpm/Makefile [hunk 1]
```diff
@@ -31,10 +31,15 @@ clean: ## remove build artifacts
 
 rpm: fedora centos ## build all rpm packages
 
-fedora: fedora-25 fedora-24 ## build all fedora rpm packages
+fedora: fedora-26 fedora-25 fedora-24 ## build all fedora rpm packages
 
 centos: centos-7 ## build all centos rpm packages
 
+fedora-26: rpmbuild/SOURCES/engine.tgz rpmbuild/SOURCES/cli.tgz ## build fedora-25 rpm packages
```
Comment (agronholm) [inline rpm/Makefile]: fedora-26, right?

Comment (andrewhsu) [inline rpm/Makefile]: fixed

**Comment by lig** [comment]
It will be very helpful to have docker repo prepared for the next Fedora version before its release. Thankfully, they have branched repo far before the release.

**Comment by eychenz** [comment]
I am wondering when can we expect the repository for F26 to be available?

**Comment by martyanov** [comment]
Not very important, but `README.md` changes are missing in the pull request.

* RPM packages for Fedora 26

Any chance it will be merged soon?

**Comment by hamidrezabstn** [comment]
Any chance it will be merged soon?

**Comment by FireFart** [comment]
In the meantime you can add this to `/etc/yum.repos.d/docker-temp.repo`:
```
[docker-ce-temp]
name=Docker CE TEMP - $basearch
baseurl=https://download.docker.com/linux/fedora/25/$basearch/stable
enabled=1
gpgcheck=1
gpgkey=https://download.docker.com/linux/fedora/gpg
```

This still works and at least let you install docker. Once the main repo is updated just delete the temp file.

**Comment by andrewhsu** [comment]
thx @seemethere for the additional updates.

LGTM

stuff works:
```
$ make ENGINE_DIR=/home/ubuntu/src/docker-ce/components/engine CLI_DIR=/home/ubuntu/src/docker-ce/components/cli fedora-26
$ find rpmbuild/ -type f -name '*.rpm' | grep RPM
rpmbuild/SRPMS/docker-ce-17.07.0-0.0.20170720.233508.git00313efde4.fc26.src.rpm
rpmbuild/RPMS/x86_64/docker-ce-17.07.0-0.0.20170720.233508.git00313efde4.fc26.x86_64.rpm
rpmbuild/RPMS/x86_64/docker-ce-debuginfo-17.07.0-0.0.20170720.233508.git00313efde4.fc26.x86_64.rpm
```

**Comment by sanseihappa** [comment]
Hi, now that this has merged, when should I expect the download site reflect F26 support? https://download.docker.com/linux/fedora/

**Comment by seemethere** [comment]
@sanseihappa it's up now: https://download.docker.com/linux/fedora/26/

**Comment by ksylvan** [comment]
At this point https://download.docker.com/linux/fedora/26/x86_64/stable/Packages/ is empty.
When will it be filled with the fc26 packages?

**Comment by aysabzevar** [comment]
I still get :
> Failed to synchronize cache for repo 'docker-ce-stable', disabling.

`[root@ops-pc-02 ~]# dnf install docker-ce`

`Failed to synchronize cache for repo 'docker-ce-stable', disabling.`

`Last metadata expiration check: 0:02:05 ago on Mon 31 Jul 2017 03:31:07 PM +0430.`

`No package docker-ce available.`

`Error: Unable to find a match`

**Comment by eychenz** [comment]
@aysabzevar , 
It's because the 'stable' and the 'edge' repo are still empty. The test repo contains the package though.
If you don't mind installing a non-stable version, go ahead and enable the test repo.
`sudo dnf config-manager --set-enabled docker-ce-test`
`sudo dnf update && sudo dnf install docker-ce`

**Comment by aysabzevar** [comment]
@eychenz 
`[root@ops-pc-02 ~]# dnf config-manager --set-enabled docker-ce-test`
`[root@ops-pc-02 ~]# dnf install docker-ce`
`Failed to synchronize cache for repo 'docker-ce-test', disabling.`
`Last metadata expiration check: 0:50:32 ago on Mon 31 Jul 2017 03:31:07 PM +0430.`
`No package docker-ce available.`
`Error: Unable to find a match`

**Comment by seemethere** [comment]
Did you try a `dnf makecache` before you tried to install?

**Comment by sanseihappa** [comment]
The documentation says `testing`, but of course, @eychenz suggested `--set-enabled docker-ce-test` works.  https://docs.docker.com/engine/installation/linux/docker-ce/fedora/#install-using-the-repository.  Perhaps the 17.07.0.ce-0.1.rc1.fc26.x86_64 package will make it into the next stable release?

**Comment by lig** [comment]
Fedora 26 is stable. Docker CE has a stable version. Still no Docker CE in the stable repo. Any timeline on this?

**Comment by RazvanPetruescu** [comment]
+1

**Comment by sanseihappa** [comment]
F27 is coming up soon, and I see that F26-stable doesn't exist (I guess F26 only gets "testing"?)
Any idea what support for docker-ce on F27 will look like? https://fedoraproject.org/wiki/Releases/27/Schedule

**Comment by sanseihappa** [comment]
F26 stable package is out there now: https://download.docker.com/linux/fedora/26/x86_64/stable/Packages/

**Pull request merged**